lib/handshake_client.ml

```diff
@@ -225,7 +225,7 @@ let answer_certificate_RSA state (session : session_data) cs raw log =
     | x -> Error (`Fatal (`NoVersions [ x ])) (* TODO: get rid of this... *)
   in
   let ver = Writer.assemble_protocol_version version in
-  let premaster = ver <+> Mirage_crypto_rng.generate 46 in
+  let premaster = ver ^ Mirage_crypto_rng.generate 46 in
```

Comment:
what we want to do: allocate 48 bytes, encode version into the first two bytes, generate_into in the remaining